python/common/org/python/types/Str.java

```diff
@@ -1764,8 +1764,36 @@ public static String _title(String input) {
     public org.python.Object upper() {
         return new org.python.types.Str(this.value.toUpperCase());
     }
-}
 
+    @org.python.Method(
```

Comment:
You need to declare the argument with this `@org.python.Method` annotation -- this maps the Python method signatures to Java arguments.
Check here for examples: http://voc.readthedocs.io/en/latest/reference/signatures.html

Reply:
Oops, don't know how I missed that. Updated the PR.

Reply:
Using `default_args` is still not the answer, a default argument means that the argument is optional. You want to use `args` instead.
You also need to test the argument handling -- that is, if it's raising the correct error when given invalid arguments.